apps/website/src/components/layout/navbar/DesktopMenu.tsx

```diff
@@ -36,13 +35,22 @@ const DropdownMenuItems: typeof Menu.Items = ({ ...props }) => (
   >
     <Menu.Items
       as="ul"
-      className="absolute right-0 top-full z-30 mt-1 flex min-w-[10rem] flex-col gap-0.5 rounded border border-black/20 bg-alveus-green-900 p-2 shadow-lg"
+      className="group absolute right-0 top-full z-30 mt-1 flex min-w-[10rem] flex-col gap-0.5 rounded border border-black/20 bg-alveus-green-900 p-2 shadow-lg"
       {...props}
     />
   </Transition>
 );
 DropdownMenuItems.displayName = "DropdownMenuItems";
 
+const DropdownMenuItem: React.FC<{ children: React.ReactElement }> = forwardRef(
+  ({ children, ...props }, ref) => (
+    <li {...props}>
+      {Children.map(children, (child) => cloneElement(child, { ref }))}
+    </li>
+  )
+);
+DropdownMenuItem.displayName = "DropdownMenuItem";
```

Comment:
This feels terrible, but it looks like the ref needs to make it all the way down to the `a` element for the keyboard navigation to work, and the native `li` element doesn't pass the ref down (duh)

Reply:
yeah not great but unless we drop headlessui there i dont see many other options